### PR TITLE
Timer fix

### PR DESCRIPTION
**Pull request by Madu86**
In this PR, I have implemented a new timing scheme to measure wall time. Legacy timing scheme (CPU timing) is still available and in order to use this instead of new, `-DREPORT_CPU_TIME` flag should be passed into fortran flag list.

### CODE REVIEW

**Comment by Madu86** [comment]
@agoetz Please make sure to merge this branch only after testing and merging AMD PR.

**Review by agoetz — APPROVED** [review]
Looking good.

**Pull request merged**
